src/components/Button/Button.tsx

```diff
@@ -1,14 +1,21 @@
 import clsx from 'clsx';
 import type { ReactNode } from 'react';
 import React, { forwardRef } from 'react';
-import ClickableStyle from '../ClickableStyle';
-import type { ClickableStyleProps, VariantStatus } from '../ClickableStyle';
+import type { Size } from '../../util/variant-types';
+import type {
+  PrimaryStatus,
+  SecondaryStatus,
+  IconStatus,
+  LinkStatus,
+} from '../ClickableStyle/ClickableStyle';
 import LoadingIndicator from '../LoadingIndicator';
 
 import styles from './Button.module.css';
 
 type ButtonHTMLElementProps = React.ButtonHTMLAttributes<HTMLButtonElement>;
 
+type ButtonVariants = IconStatus | PrimaryStatus | SecondaryStatus | LinkStatus;
```

Comment:
Eventually, we won't need the variant/status types to be this granular and coupled in `Button`/`Link`. At next major, this will be simplified to be the individual statuses and variants, and we can then drop some of this complexity (or sequester it so just `ClickableStyle`